src/Microsoft.TemplateEngine.Cli/Components.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Cli.PostActionProcessors;
+using Microsoft.TemplateEngine.Cli.TemplateSearch;
+using Microsoft.TemplateSearch.Common;
+
+namespace Microsoft.TemplateEngine.Cli
+{
+    public static class Components
+    {
+        private static readonly AddProjectsToSolutionPostAction AddProjectsToSolutionPostAction = new AddProjectsToSolutionPostAction();
+        private static readonly AddReferencePostActionProcessor AddReferencePostActionProcessor = new AddReferencePostActionProcessor();
+        private static readonly DotnetRestorePostActionProcessor DotnetRestorePostActionProcessor = new DotnetRestorePostActionProcessor();
+
+        public static IReadOnlyList<(Type Type, IIdentifiedComponent Instance)> AllComponents { get; } =
+            new (Type Type, IIdentifiedComponent Instance)[]
+            {
+                (typeof(IPostActionProcessor), AddProjectsToSolutionPostAction),
```

Comment:
I think we need to make these interfaces public, otherwise it might be unusable.

Reply:
Its not less usable than it is today when we do `RegisterMany(assembly.GetTypes())`... Host kind of doesn't need to know anything about components, if they want to disable specific component they can do it via `Instance.Id`...